Add prereq, install page, upgrade (#1088)
kubernetes-sigs/dra-driver-nvidia-gpu: docs/install.md

If you installed with `--set resources.computeDomains.enabled=false`, the `controller` pod will not be present and the `kubelet-plugin` pod will show `1/1`.

2. Confirm the DeviceClasses were registered:

```bash
kubectl get deviceclass
```

Example output:

```
NAME                                      AGE
compute-domain-daemon.nvidia.com           1m
compute-domain-default-channel.nvidia.com  1m
gpu.nvidia.com                             1m
mig.nvidia.com                             1m
vfio.gpu.nvidia.com                        1m
```

`gpu.nvidia.com` is used for standard GPU allocation. `mig.nvidia.com` and `vfio.gpu.nvidia.com` are registered but only usable with the appropriate hardware and configuration. The `compute-domain-*` classes are used by the ComputeDomain controller.

If you installed with only ComputeDomain support, `gpu.nvidia.com`, `mig.nvidia.com`, `vfio.gpu.nvidia.com` will not be installed.

If you installed with only GPU allocation support, `compute-domain-daemon.nvidia.com`, `compute-domain-default-channel.nvidia.com` will not be installed.

3. Confirm GPU nodes have advertised their ResourceSlices:

```bash
kubectl get resourceslice -o wide
```

> When GPU allocation support is enabled, each GPU node should appear with a slice listing its available devices. If no slices appear, the kubelet plugin is not communicating with the API server.

Reviewer: its imp to call this out but note if this actually happens, then there is a bug either in the driver or mistake in install step

## Optional: Configure Helm

The following parameters are most commonly set at install time.

| Parameter | Default | Description |
|---|---|---|
| `nvidiaDriverRoot` | `/` | Path to the GPU driver root on the host. If the NVIDIA GPU Operator manages the NVIDIA GPU driver on your nodes, use `/run/nvidia/driver`, the default location for Operator managed drivers. For GKE, use `/home/kubernetes/bin/nvidia` Incorrect values are a common source of error. |
| `resources.gpus.enabled` | `true` | Enable the GPU kubelet plugin. Requires `gpuResourcesEnabledOverride=true`. |
| `resources.computeDomains.enabled` | `true` | Enable the ComputeDomain controller and kubelet plugin. |
| `gpuResourcesEnabledOverride` | `false` | Required to enable GPU allocation resources. |

> | `featureGates` | `{}` | Map of feature gate name to boolean. See [Feature gates](reference/feature-gates.md). |

Reviewer: should we have the table of FGs?

Author: I have a table in a separate PR, https://github.com/kubernetes-sigs/dra-driver-nvidia-gpu/pull/1098

| `logVerbosity` | `4` | Log verbosity level (0–7). Higher values produce more output. |

To list all available parameters:

```bash
helm show values nvidia/nvidia-dra-driver-gpu
```

---

## Optional: Admission webhook

> The admission webhook validates opaque configuration in `ResourceClaim` and `ResourceClaimTemplate` specs, providing early feedback on invalid values. It is disabled by default.

Reviewer: pls link `opaque configuration`

Author: i added a link below

Prerequisite: [cert-manager](https://cert-manager.io/) must be installed in your cluster.

1. Install cert-manager:

```bash
helm install \
    --repo https://charts.jetstack.io \
    --version v1.16.3 \
    --create-namespace \
    --namespace cert-manager \
    --wait \
    --set crds.enabled=true \
    cert-manager \
    cert-manager
```

2. Enable the webhook:

```bash
helm install nvidia-dra-driver-gpu nvidia/nvidia-dra-driver-gpu \
    --version=25.12.0 \
    --create-namespace \
    --namespace nvidia-dra-driver-gpu \
    --set gpuResourcesEnabledOverride=true \
    --set webhook.enabled=true
```

To use a pre-existing TLS secret instead of cert-manager, set `webhook.tls.mode=secret` and provide `webhook.tls.secret.name` and `webhook.tls.secret.caBundle`.

> ## Run a sample GPU allocation workload

Reviewer: pls add a note we have many examples in the demo folder

# Install

This page walks through installing the DRA Driver for NVIDIA GPUs and validating that GPU or ComputeDomain allocation is working correctly on your cluster.

Before starting, make sure all [prerequisites](prerequisites.md) are met.

> If you have the NVIDIA GPU Operator installed, its recommended that you following the [GPU Operator install guide](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/dra-intro-install.html) instead.

Reviewer 2: ```suggestion
If you have the NVIDIA GPU Operator installed, it's recommended that you follow the [GPU Operator install guide](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/dra-intro-install.html) instead.
```

---

## Install

1. Add the Helm repository:

```bash
helm repo add nvidia https://helm.ngc.nvidia.com/nvidia && helm repo update
```

> 2. Install the driver. The following commands install the DRA Driver with GPU allocation and ComputeDomain support. If you only plan to use one of the resource plugins, the other can be safely ignored without interference on your cluster. You can choose to disbale either plugin by including `--set resources.computeDomains.enabled=false` to disable ComputeDomain support, or `--set resources.gpus.enabled=false` and remove `--set gpuResourcesEnabledOverride=true` from the Helm command.

Reviewer 2: ```suggestion
2. Install the driver. The following commands install the DRA Driver with GPU allocation and ComputeDomain support. If you only plan to use one of the resource plugins, the other can be safely ignored without interference on your cluster. You can choose to disable either plugin by including `--set resources.computeDomains.enabled=false` to disable ComputeDomain support, or `--set resources.gpus.enabled=false` and remove `--set gpuResourcesEnabledOverride=true` from the Helm command.
```

>Note, if you are installing on GKE, include `--set nvidiaDriverRoot=/home/kubernetes/bin/nvidia` to have the driver use the default NVIDIA Driver install path on GKE.

```bash

> helm install nvidia-dra-driver-gpu nvidia/nvidia-dra-driver-gpu \

Reviewer 2: ```suggestion
helm install dra-driver-nvidia-gpu oci://registry.k8s.io/dra-driver-nvidia/charts/dra-driver-nvidia-gpu \
```

>     --version=25.12.0 \

Reviewer 2: ```suggestion
    --version=0.4.0 \
```

Author: updated!

    --create-namespace \

>     --namespace nvidia-dra-driver-gpu \

Reviewer 2: ```suggestion
    --namespace dra-driver-nvidia-gpu \
```

Reviewer 2: This recurs throughout!

```bash
helm show values nvidia/nvidia-dra-driver-gpu
```

---

## Optional: Admission webhook

The admission webhook validates opaque configuration in `ResourceClaim` and `ResourceClaimTemplate` specs, providing early feedback on invalid values. It is disabled by default.
Refer to the [API reference](reference/api.md) page for more details on this configuration.

Prerequisite: [cert-manager](https://cert-manager.io/) must be installed in your cluster.

1. Install cert-manager:

```bash
helm install \
    --repo https://charts.jetstack.io \
    --version v1.16.3 \
    --create-namespace \
    --namespace cert-manager \
    --wait \
    --set crds.enabled=true \
    cert-manager \
    cert-manager
```

2. Enable the webhook:

```bash
helm install nvidia-dra-driver-gpu nvidia/nvidia-dra-driver-gpu \
    --version=25.12.0 \
    --create-namespace \
    --namespace nvidia-dra-driver-gpu \
    --set gpuResourcesEnabledOverride=true \
    --set webhook.enabled=true
```

To use a pre-existing TLS secret instead of cert-manager, set `webhook.tls.mode=secret` and provide `webhook.tls.secret.name` and `webhook.tls.secret.caBundle`.

## Run a sample GPU allocation workload

> This section inlcude steps for deploying a sample application for GPU allocation and ComputeDomains on your cluster.

Reviewer 2: ```suggestion
This section includes steps for deploying a sample application for GPU allocation and ComputeDomains on your cluster.
```

# Install

Install the DRA Driver for NVIDIA GPUs and validate that GPU or ComputeDomain allocation is working on your cluster.

Before you begin:

- Confirm all [prerequisites](prerequisites.md) are met.
- If you have the NVIDIA GPU Operator installed, follow the [GPU Operator install guide](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/dra-intro-install.html) instead.

---

## Install the chart

The following command installs the DRA Driver with both GPU allocation and ComputeDomain support enabled.

By default, both resource plugins are enabled. If you only need one, the other can be left enabled with no impact on the cluster, or you can disable it explicitly:

- To disable ComputeDomain support, add `--set resources.computeDomains.enabled=false`.

> - To disable GPU allocation, add `--set resources.gpus.enabled=false` and remove `--set gpuResourcesEnabledOverride=true`.

Reviewer 3: nit: Removing `gpuResourcesEnabledOverride=true` is not required if GPU plugin is not enabled.

Author: removed.
